python/cudf/cudf/tests/test_string.py

```diff
@@ -1812,7 +1812,33 @@ def test_string_count(data, pat, flags):
     assert_eq(as_index(gs).str.count(pat=pat), pd.Index(ps).str.count(pat=pat))
 
 
-def test_string_findall():
+@pytest.mark.parameterize(
+    "pat, flags",
+    [
+        ("Monkey", 0),
+        ("on", 0),
+        ("b", 0),
+        ("on$", 0),
+        ("on$", re.MULTILINE),
+        ("o.*k", re.DOTALL),
+    ],
+)
+def test_string_findall(pat, flags):
+    test_data = ["Lion", "Monkey", "Rabbit", "Don\nkey"]
+    ps = pd.Series(test_data)
+    gs = cudf.Series(test_data)
+
+    # TODO: Update this test to remove "expand=False" when removing the expand
+    # parameter from Series.str.findall.
+    assert_eq(
+        ps.str.findall(pat, flags), gs.str.findall(pat, flags, expand=False)
+    )
+
+
+@pytest.mark.filterwarnings("ignore:The expand parameter is deprecated")
+def test_string_findall_expand_True():
```

Comment:
@galipremsagar Good catch. I wasn't thinking about this correctly. 🙃 I have added tests for both expand=True (the default behavior and previous/current working behavior) and expand=False (the future behavior, which was fixed in this PR).